include/LeMonADE/updater/UpdaterAbstractCreate.h

```diff
@@ -0,0 +1,301 @@
+  bool add_monomer_to_parent(uint32_t parent_id, int32_t type=1);
+  
+  //! function to add a monomer at a spezific place
+  bool add_monomer_to_position(VectorInt3 pos, int32_t type=1);
+  
+  //! function to add a new monomer inbetween two others
+  bool add_monomer_inside_connected_pair(uint32_t indexA, uint32_t indexB, int32_t type=1);
+  
+  //! function to move the whole system
+  void move_system(int32_t nsteps);
+  
+  //! function to find groups of connected monomers
+  void linearize_system();
+  
+};
+
+/**
+* The initialize function handles the new systems information.
+*
+* @tparam IngredientsType Features used in the system. See Ingredients.
+*/
+template < class IngredientsType >
+void UpdaterAbstractCreate<IngredientsType>::initialize(){
+  
+}
+
+/**
+* Execution of the system creation
+*
+* @tparam IngredientsType Features used in the system. See Ingredients.
+*/
+template < class IngredientsType >
+bool UpdaterAbstractCreate<IngredientsType>::execute(){
+  
+}
+
+/**
+* Standard clean up.
+*
+* @tparam IngredientsType Features used in the system. See Ingredients.
+*/
+template < class IngredientsType >
+void UpdaterAbstractCreate<IngredientsType>::cleanup(){
+  
+}
+
+/******************************************************************************/
+/**
+ * @brief function to add a standalone monomer
+ * @param type attribute tag of the new monomer
+ * @return <b false> if position is not free, <b true> if move was applied
+ */
+template<class IngredientsType>
+bool UpdaterAbstractCreate<IngredientsType>::add_lonely_monomer(int32_t type){
+  // set properties of add Monomer Move
+  MoveAddScMonomer addmove;
+  addmove.init(ingredients);
+  
+  int32_t counter(0);
+  while(counter<10000){
+    VectorInt3 newPosition((rng.r250_rand32() % (ingredients.getBoxX()-1)),
+				  (rng.r250_rand32() % (ingredients.getBoxY()-1)),
+				  (rng.r250_rand32() % (ingredients.getBoxZ()-1)));
+    addmove.setPosition(newPosition);
+    if(addmove.check(ingredients)==true){
+      addmove.setType(type);
+      addmove.apply(ingredients);
+      return true;
+    }
+    counter++;
+  }
+  return false;
+}
+
+/******************************************************************************/
+/**
+ * @brief function to add a monomer to a parent monomer
+ * @param parent_id id of monomer to connect with
+ * @param type attribute tag of the new monomer
+ * @return <b false> if position is not free, <b true> if move was applied
+ */
+template<class IngredientsType>
+bool UpdaterAbstractCreate<IngredientsType>::add_monomer_to_parent(uint32_t parent_id, int32_t type){
+  // set properties of add Monomer Move
+  MoveAddScMonomer addmove;
+  addmove.init(ingredients);
+  
+  int32_t counter(0);
+  
+  while(counter<10000){
+    //try at most 30 random bondvectors to find a new monomer position
+    for(uint i=0;i<30;i++){
+      // get id of random bondvector with index <= 22, also P(2,0,0)
+      std::size_t randBondVectorID((rng.r250_rand32() % 6)+17);
```

Comment:
not sure about this way of generating the bond vector. issues are

- is this also valid in bcc bfm?
- it seems it now depends on the naming and numbering of the bonds, which is in principle arbitrary. Maybe we can just generate a vector from P(2,0,0) directly, without getting it from the bondset? In this case one would of course have to check if that vector is  part of the bond set at all

Or any other suggestions?